core/config.js

```diff
@@ -2,42 +2,73 @@
 const fs = require('fs-extra')
 const _ = require('lodash')
 const path = require('path')
+const os = require('os')
 
 const { hideOnWindows } = require('./utils/fs')
 const logger = require('./logger')
 
 const log = logger({
   component: 'Config'
 })
+const TMP_CONFIG_PATH = path.join(os.tmpdir(), 'cozy-desktop', 'config.json')
+
+function writeTmpConfig (config) {
+  fs.ensureFileSync(TMP_CONFIG_PATH)
+  fs.writeFileSync(TMP_CONFIG_PATH, config)
+}
+
+function moveTmpConfig (configPath) {
+  fs.copySync(TMP_CONFIG_PATH, configPath, fs.constants.COPYFILE_FICLONE)
+  fs.unlinkSync(TMP_CONFIG_PATH)
+}
 
 // Config can keep some configuration parameters in a JSON file,
 // like the devices credentials or the mount path
 module.exports = class Config {
   // Create config file if it doesn't exist.
   constructor (basePath) {
     this.configPath = path.join(basePath, 'config.json')
+    fs.ensureFileSync(this.configPath)
     this.dbPath = path.join(basePath, 'db')
     fs.ensureDirSync(this.dbPath)
     hideOnWindows(basePath)
-    fs.ensureFileSync(this.configPath)
 
-    if (fs.readFileSync(this.configPath).toString() === '') {
-      this.reset()
+    this.config = Config.import(this.configPath)
+    this.persist()
+
+    // FIXME: autoBind(this)
+  }
+
+  // Import the configuration from disk
+  static import (configPath) {
+    let config
+
+    if (fs.existsSync(TMP_CONFIG_PATH)) {
+      config = this.safeLoad(TMP_CONFIG_PATH)
+
+      if (!!config && !!config.configPath) {
+        moveTmpConfig(config.configPath)
```

Comment:
:warning: this can throw (for instance if the user does not have permission to write `config.configPath`) and leave the app unusable.

Reply:
Idem for persist.

Reply:
Actually, I think there's a more important issue here. Writing the file to an unknown location could be used to overwrite completely unrelated files and be used for malicious purposes.
I'll think about what can be done here to make sure a config path change is made only at the user's request.